src/plugins/data_views/public/services/has_data.ts

```diff
@@ -77,6 +82,41 @@ export class HasData {
     return source;
   };
 
+  private getIndicesViaSearch = async ({
+    http,
+    pattern,
+    showAllIndices,
+  }: {
+    http: HttpStart;
+    pattern: string;
+    showAllIndices: boolean;
+  }): Promise<boolean> =>
+    http
+      .post<IndicesViaSearchResponse>(`/internal/search/ese`, {
+        body: JSON.stringify({
+          params: {
+            ignore_unavailable: true,
+            expand_wildcards: showAllIndices ? 'all' : 'open',
+            index: pattern,
+            body: {
+              size: 0, // no hits
+              aggs: {
+                indices: {
+                  terms: {
+                    field: '_index',
+                    size: 200,
+                  },
+                },
+              },
+            },
+          },
+        }),
+      })
+      .then((resp) => {
+        return !!(resp && resp.total >= 0);
+      })
+      .catch(() => false);
```

Comment:
Should we assume that "data exists" in case of error instead? as discussed in the issue https://github.com/elastic/kibana/issues/132478#issuecomment-1131958539

Reply:
I would not as it could be any error and that would triiger it to be true. I think this should fix the issue for now and we will wait for the permission issue to be fixed.